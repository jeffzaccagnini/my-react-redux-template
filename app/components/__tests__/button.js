import { Simulate, renderIntoDocument } from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import Button from 'rk/components/button';

describe('components/button', function() {
    it('should render the button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(<Button label="MyButton" onClick={onClickHandler} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with class name', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(<Button className="my-class" label="MyButton" onClick={onClickHandler} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with disabled', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(<Button disabled={true} label="MyButton" onClick={onClickHandler} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should trigger onClick handler', function(done) {
        const onClickHandler = () => done();
        const component = renderIntoDocument(<Button label="MyButton" onClick={onClickHandler} />);
        Simulate.click(component.refs.button);
    });
});
