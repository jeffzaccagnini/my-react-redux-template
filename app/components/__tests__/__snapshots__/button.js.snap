exports[`components/button should render the button 1`] = `
<button
  className="sbm-button"
  onClick={[Function]}
  type="button">
  MyButton
</button>
`;

exports[`components/button should render with class name 1`] = `
<button
  className="sbm-button my-class"
  onClick={[Function]}
  type="button">
  MyButton
</button>
`;

exports[`components/button should render with disabled 1`] = `
<button
  className="sbm-button"
  onClick={[Function]}
  type="button">
  MyButton
</button>
`;
